tldw_Server_API/app/core/Audit/unified_audit_service.py

```
@@ -1641,10 +1655,11 @@ async def _copy_rows() -> None:
                             "pii_detected": bool(data.get("pii_detected") or False),
                             "compliance_flags": _json_text(data.get("compliance_flags"), default="[]"),
                             "metadata": _json_text(data.get("metadata"), default="{}"),
+                            "chain_hash": str(data.get("chain_hash") or ""),
                         }
                         if self._shared_mode:
                             record["tenant_user_id"] = self._resolve_tenant_id_for_write(
-                                raw_tenant=None,
+                                raw_tenant=tenant_user_id_override,
```

Ruff (0.15.9)

[error] Undefined name `tenant_user_id_override`

(F821)